icssc-projects/AntAlmanac
# Add navigation warning when there are unsaved changes

### Summary
Any of the following actions will set the `unsavedChanges` variable to true:
Adding a course, adding a section, deleting a course/custom event, undoing a delete, clearing a schedule, changing an event color, editing a custom event, and copying a schedule

If the user attempts to leave the site, they will get a navigation warning
![navigation-warning](https://user-images.githubusercontent.com/29494270/100067784-88c35800-2deb-11eb-9547-cde71022aaa5.gif)

If a user attempts to load a new schedule, they will receive a confirm prompt notifying them of unsaved changes.
![image](https://user-images.githubusercontent.com/29494270/100067893-ac869e00-2deb-11eb-9943-77876137ba1a.png)

`unsavedChanges` is set back to false after a save or load.

### Test Plan
- Verify that attempting to navigate away will trigger a warning after one of those actions is performed. _Make sure to refresh the page between each action._
- Verify that confirm prompt appears when attempting to load a new schedule while there are unsaved changes.
- Verify that no warning appears after a save, after a load, or if none of those actions are performed.

### Issues
Resolves #7
